server/routerlicious/packages/services/src/mongoExceptionRetryRules/mongoErrorRetryAnalyzer.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Lumberjack } from "@fluidframework/server-services-telemetry";
+import { DefaultRule } from "./defaultRule";
+import { IMongoExceptionRetryRule } from "./IMongoExceptionRetryRule";
+import { mongoErrorRetryRuleset } from "./mongoError";
+import { mongoNetworkErrorRetryRuleset } from "./mongoNetworkError";
+
+export class MongoErrorRetryAnalyzer {
+    public static shouldRetry(error: Error): boolean {
+        const rule = MongoErrorRetryAnalyzer.getRetryRule(error);
+        if (!rule) {
+            // This should not happen.
+            Lumberjack.error("MongoErrorRetryAnalyzer.shouldRetry() didn't get a rule", undefined, error);
+            return false;
+        }
+
+        return rule.shouldRetry;
+    }
+
+    static getRetryRule(error: Error): IMongoExceptionRetryRule {
+        if (error.name === "MongoNetworkError") {
+            return MongoErrorRetryAnalyzer.getRetryRuleFromSet(error, mongoErrorRetryRuleset);
+        }
+
+        if (error.name === "MongoError") {
+            return MongoErrorRetryAnalyzer.getRetryRuleFromSet(error, mongoNetworkErrorRetryRuleset);
```

Comment:
I think these are flipped around. name==MongoError is using the network error ruleset, and networkError is using the mongoError ruleset